libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,336 @@
 package com.baeldung.jnats;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
-
-import org.junit.Test;
-
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+
+import io.nats.client.Connection;
+import io.nats.client.Dispatcher;
+import io.nats.client.ErrorListener;
+import io.nats.client.JetStreamStatusException;
 import io.nats.client.Message;
-import io.nats.client.SyncSubscription;
-
+import io.nats.client.Nats;
+import io.nats.client.Options;
+import io.nats.client.Subscription;
+
+/**
+ * All the tests in this class require that a NATS server be running on localhost at the default port.
+ * See {@link <a href="https://docs.nats.io/nats-server/installation">Installing a NATS Server</a>}.
+ * <p>
+ * IMPORTANT: Awaitility.await is used to simulate the real behavior
```

Comment:
We aren't really simulating anything here, we are literally waiting for the message to arrive
```suggestion
 * IMPORTANT: Awaitility.await is used to account for the real behavior
```

Reply:
fixed